pipelines/context.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package pipelines
+
+import (
+	"context"
+)
+
+type contextKey struct{}
+
+var activePathwayKey = contextKey{}
+
+// ToContext returns a copy of the given context which includes the pathway p.
+func ToContext(ctx context.Context, p Pathway) context.Context {
+	return context.WithValue(ctx, activePathwayKey, p)
+}
+
+// FromContext returns the pathway contained in the given context.
+func FromContext(ctx context.Context) (p Pathway, ok bool) {
+	if ctx == nil {
+		return Pathway{}, false
+	}
+	v := ctx.Value(activePathwayKey)
+	if s, ok := v.(Pathway); ok {
+		return s, true
+	}
+	return Pathway{}, false
+}
+
+// SetCheckpoint sets a checkpoint on the pathway in the context.
+// if there is no pathway in the context, it creates a new one.
+func SetCheckpoint(ctx context.Context, edge string) (Pathway, context.Context) {
```

Comment:
There is no documentation explaining what `edge` is? What is "edge"? Is it the name of the checkpoint? If yes, why is it called "edge"? Why not "name" or "point"?